tests/kernel/device/src/main.c

```diff
@@ -474,8 +470,11 @@ void test_main(void)
 			 ztest_unit_test(test_dummy_device),
 			 ztest_unit_test(test_enable_and_disable_automatic_idle_pm),
 			 ztest_unit_test(test_pre_kernel_detection),
+			 ztest_unit_test(test_bogus_dynamic_name),
```

Comment:
this should not needed, if userspace is not enabled, `ztest_user_unit..` behaves like  `ztest_unit...`

Reply:
Yes, but the test is never run without userspace, is it?

Reply:
No it is not, but the fix should be in the testcase itself (which you did), not here.

Reply:
agreed

Reply:
The test uses `CONFIG_TEST_USERSPACE=y` unconditionally.  AFAICT that implies `CONFIG_USERSPACE`.  The only reason the test passed at all was because it was conditional on `CONFIG_USERSPACE`, and the validation that made it work was only performed with `CONFIG_USERSPACE=y`.

So to catch the original error, it would have been necessary to both eliminate the source-code filtering of the test with `CONFIG_USERSPACE=n`, *and* to build the code with `CONFIG_USERSPACE=n`.

So from what I can tell we would need to modify this test (and potentially others) to build and run without userspace in order to catch future instances of this sort of problem.

Reply:
> The test uses `CONFIG_TEST_USERSPACE=y` unconditionally. AFAICT that implies `CONFIG_USERSPACE`.

all of this depends on what the platform can do, the same test can run without userspace on any platform.